pkg/pillar/kubeapi/vitoapiserver.go

```diff
@@ -313,6 +313,92 @@ func RolloutDiskToPVC(ctx context.Context, log *base.LogObject, exists bool,
 	return nil
 }
 
+// GetPVFromPVC : Returns volume name (PV) from the PVC name
+func GetPVFromPVC(pvcName string, log *base.LogObject) (string, error) {
+	// Get the Kubernetes clientset
+	clientset, err := GetClientSet()
+	if err != nil {
+		err = fmt.Errorf("failed to get clientset: %v", err)
+		log.Error(err)
+		return "", err
+	}
+	// Get the PersistentVolumeClaim (PVC)
+	pvc, err := clientset.CoreV1().PersistentVolumeClaims(EVEKubeNameSpace).Get(context.TODO(), pvcName, metav1.GetOptions{})
+	if err != nil {
+		err = fmt.Errorf("Error fetching PersistentVolumeClaim %s: %v", pvcName, err)
+		log.Error(err)
+		return "", err
+	}
+
+	// Get the associated PersistentVolume (PV) name
+	volumeName := pvc.Spec.VolumeName
+
+	log.Noticef("PersistentVolume %s is associated with PVC %s", volumeName, pvcName)
+
+	return volumeName, nil
+}
+
+// GetVolumeAttachmentFromPV : Return volume attachment if any for that PV along with nodename.
+// longhorn attaches to the PV to serve to the apps. This API returns the attachment name and nodename.
+// We use that attachment name to delete the attachment during failover.
+// Basically the attachment of previous node needs to be deleted to attach to current node.
+func GetVolumeAttachmentFromPV(volName string, log *base.LogObject) (string, string, error) {
+	// Get the Kubernetes clientset
+	clientset, err := GetClientSet()
+	if err != nil {
+		err = fmt.Errorf("failed to get clientset: %v", err)
+		log.Error(err)
+		return "", "", err
+	}
+	// List all VolumeAttachments and find the one corresponding to the PV
+	volumeAttachments, err := clientset.StorageV1().VolumeAttachments().List(context.TODO(), metav1.ListOptions{})
```

Comment:
Is it a synchronous call to the API? Are we sure we will not have issues with the watchdog in this case if the requests are handled longer than expected? This is a generic question for all API calls here.

Reply:
Its is synchronous call, but I do not anticipate it takes even a second to get the result. Number of volumes we have in our deployments are in single digits.

Reply:
Ok, and what about the other API calls in the file? Like `.Delete()`... So we are pretty sure all of them run fast enough, yep?

Reply:
Also Kubernetes API calls execution are asynchronous by default. So only sync call is to issue the command and return back after that any action will happen in background.

Reply:
Any more questions ? Or can this be merged ?